### ISSUE TITLE
Test PyPI failing due to accidental publishes from another branch

### ISSUE THREAD

**Issue by JessicaS11** [icesat2py/icepyx#562]
Since the transition to pyproject.toml(#539), we routinely get failures of the testPyPI action after it's merged into development (e.g. [here](https://github.com/icesat2py/icepyx/actions/runs/10376145520/job/28727399601?pr=561)):

```
 Uploading icepyx-1.1.1.dev11-py3-none-any.whl
WARNING  Error during upload. Retry with the --verbose option for more details. 
ERROR    HTTPError: 400 Bad Request from https://test.pypi.org/legacy/          
         File already exists. See https://test.pypi.org/help/#file-name-reuse   
         for more information.  
```

Anyone know why these are happening (even after #541)?

**Comment by mfisher87** [comment]
>Uploading icepyx-1.1.1.dev11-py3-none-any.whl ... File already exists

The file for dev11 was uploaded on August 12 -- our CI should be trying to upload dev12 instead, correct?

**Comment by mfisher87** [comment]
Would you like to have a zoom chat tomorrow? I know you're preparing for the hack week and I want to take as much load off your plate on this as I can. Apologies about the inconvenience of whatever it was we missed here!

**Comment by weiji14** [comment]
Hmm, I'm looking at the `publish_to_pypi.yml` and see that the trigger should only be on pushes to the `development` branch, on `release`, or `workflow_dispatch`:

https://github.com/icesat2py/icepyx/blob/c85baec810b439618b74c7147d4911033c7b840f/.github/workflows/publish_to_pypi.yml#L4-L13

But the Actions tab is showing the workflow running on both development and v2 branches?

![image](https://github.com/user-attachments/assets/8d1f158b-fda4-4303-b5c5-bd161cbdeae2)

Did someone trigger the GitHub Action manually on the `v2` branch? That could be why there were duplicate `.devXX` TestPyPI builds.

**Comment by mfisher87** [comment]
Ahh shoot. Thanks for figuring that out! That was my mistake. I deliberately set some actions to run on the v2 branch, because it will be long-running, but I should _not_ have done so with this action.

**Comment by mfisher87** [comment]
Yes, I did that immediately when @weiji14 realized my mistake. Should be good going forward! Apologies all :grimacing:

**Comment by mfisher87** [comment]
I think we're good to close this as resolved
